src/library/librarycontrol.h

```diff
@@ -65,27 +79,51 @@ class LibraryControl : public QObject {
     void slotDecrementFontSize(double v);
 
   private:
-    Library* m_pLibrary;
-
-    ControlObject* m_pSelectNextTrack;
-    ControlObject* m_pSelectPrevTrack;
-    ControlObject* m_pSelectTrack;
-
-    ControlObject* m_pSelectSidebarItem;
-    ControlObject* m_pSelectPrevSidebarItem;
-    ControlObject* m_pSelectNextSidebarItem;
-
-    ControlObject* m_pToggleSidebarItem;
-    ControlObject* m_pLoadSelectedIntoFirstStopped;
-    ControlObject* m_pAutoDjAddTop;
-    ControlObject* m_pAutoDjAddBottom;
-
-    ControlObject* m_pFontSizeKnob;
-    ControlPushButton* m_pFontSizeIncrement;
-    ControlPushButton* m_pFontSizeDecrement;
-
-    WLibrary* m_pLibraryWidget;
-    WLibrarySidebar* m_pSidebarWidget;
+    Library* m_pLibrary = nullptr;
+
+    // Simulate pressing a key on the keyboard
+    void emitKeyEvent(QKeyEvent&& event);
+
+    // Controls to navigate vertically within currently focussed widget (up/down buttons)
+    ControlPushButton* m_pMoveUp = nullptr;
```

Comment:
The de-facto rule is "initialize every member in the member initializer list". ControlObjects would be the most common exception. I don't think the Qt object tree bit is relevant to this particular discussion since you can easily create objects that are part of the tree with the member initializer list since you can pass "this" as the parent if you choose.

The style guide is here:
http://www.mixxx.org/wiki/doku.php/coding_guidelines
but it does not say anything beyond the Google C++ style guide about member variable initialization.